e2e_test/batch/basic/lookup_join.slt.part

```diff
@@ -47,6 +47,13 @@ select * from t1 right join t3 on t1.v1 = t3.v1 order by t1.v1, t1.v2;
 1 4 1 2
 NULL NULL 4 1
 
+# A bug in https://github.com/risingwavelabs/risingwave/pull/12527
+# lookup join panics when the scan is single distribution
+query I
+select 1 from (select * from t1 left join t3 on t1.v1 = t3.v1 order by t1.v1, t1.v2) where exists (select 1) limit 1;
+----
+1
```

Comment:
Could you please elaborate more about this case about lookup join panic?

Reply:
Basically, I used `self.clone()` in `two_phase_limit` instead of `self.clone_with_input`, so its input is the one before `to_distributed` is called: `LocalLookupJoin` + `BatchScan` (Single dist).

What's interesting is that without `where exists (select 1)` the result is correct.

```
dev=> explain (verbose) select 1 from (select * from t1 left join t3 on t1.v1 = t3.v1 order by t1.v1, t1.v2) where exists (select 1) limit 1;
                                    QUERY PLAN
-----------------------------------------------------------------------------------
 BatchExchange { order: [], dist: Single }
 └─BatchProject { exprs: [1:Int32] }
   └─BatchLimit { limit: 1, offset: 0 }
     └─BatchNestedLoopJoin { type: LeftSemi, predicate: true, output: all }
       ├─BatchLookupJoin { type: LeftOuter, predicate: t1.v1 = t3.v1, output: [] }
       │ └─BatchScan { table: t1, columns: [t1.v1], distribution: Single }
       └─BatchValues { rows: [[]] }
(7 rows)

dev=>
dev=> explain (verbose) select 1 from (select * from t1 left join t3 on t1.v1 = t3.v1 order by t1.v1, t1.v2) limit 1;
                                    QUERY PLAN
-----------------------------------------------------------------------------------
 BatchProject { exprs: [1:Int32] }
 └─BatchLimit { limit: 1, offset: 0 }
   └─BatchExchange { order: [], dist: Single }
     └─BatchLimit { limit: 1, offset: 0 }
       └─BatchLookupJoin { type: LeftOuter, predicate: t1.v1 = t3.v1, output: [] }
         └─BatchExchange { order: [], dist: UpstreamHashShard(t1.v1) }
           └─BatchScan { table: t1, columns: [t1.v1], distribution: SomeShard }
(7 rows)
```